Rewrite doc, eliminating redundancies. (#520)
PennyLaneAI/pennylane-lightning: README.rst

+===========+=========+========+=============+================+=================+================+
| Linux x86 | pip     | pip    | source      | pip            | source          | source         |
+-----------+---------+--------+-------------+----------------+-----------------+----------------+
| Linux ARM | pip     | source |             | pip            | source          | source         |
+-----------+---------+--------+-------------+----------------+-----------------+----------------+
| Linux PPC | pip     | source |             | pip            | source          | source         |
+-----------+---------+--------+-------------+----------------+-----------------+----------------+
| MacOS x86 | pip     |        |             | pip            |                 |                |
+-----------+---------+--------+-------------+----------------+-----------------+----------------+
| MacOS ARM | pip     |        |             | pip            |                 |                |
+-----------+---------+--------+-------------+----------------+-----------------+----------------+
| Windows   | pip     |        |             |                |                 |                |

> +-----------+---------+--------+-------------+----------------+-----------------+----------------+

Reviewer: Useful table!

Reviewer 2: We can update for conda/spack in a follow-up PR.

.. installation_LQubit-start-inclusion-marker-do-not-remove

************
Installation
************

Pip installation
================

PyPI wheels
-----------

Lightning plugins can be installed using ``pip`` as follows

.. code-block:: console

    $ pip install pennylane-lightning

The above command will install the Lightning-Qubit plugin (the default since it is most broadly supported).
Other backends are installed with

.. code-block:: console

    $ pip install pennylane-lightning[gpu]

>     $ pip install pennylane-lightning[kokkos]

Reviewer 3: This information is redundant to the one provided in the specific sections. Do we want that?

Author: I repeated it briefly in the specific sections in case someone jumps right there without going through the generic section. I don't mind keeping or removing it.

Reviewer 2: Let's tell them what we told them --- being overly explicit here helps.

> Other backends are installed with
> 
> .. code-block:: console
> 
>     $ pip install pennylane-lightning[gpu]
>     $ pip install pennylane-lightning[kokkos]
> 
> which will install the Lightning-GPU and Lightning-Kokkos (OpenMP) backends respectively.

Reviewer 3: ```suggestion
In order to install the Lightning-GPU and Lightning-Kokkos (OpenMP) backends, you can respectively use the following commands:

.. code-block:: console

    $ pip install pennylane-lightning[gpu]
    $ pip install pennylane-lightning[kokkos]

```

One can activate simultaneously one serial, one parallel CPU host (e.g. "OPENMP", "THREADS") and one parallel GPU device backend (e.g. "HIP", "CUDA"), but not two of any category at the same time.
For "HIP" and "CUDA", the appropriate software stacks are required to enable compilation and subsequent use.
Similarly, the CMake option ``-DKokkos_ARCH_{...}=ON`` must also be specified to target a given architecture.
A list of the architectures is found on the `Kokkos wiki <https://github.com/kokkos/kokkos/wiki/Macros#architectures>`_.
Note that "THREADS" backend is not recommended since `Kokkos <https://github.com/kokkos/kokkos-core-wiki/blob/17f08a6483937c26e14ec3c93a2aa40e4ce081ce/docs/source/ProgrammingGuide/Initialization.md?plain=1#L67>`_ does not guarantee its safety.

.. installation_LKokkos-end-inclusion-marker-do-not-remove

Please refer to the `plugin documentation <https://docs.pennylane.ai/projects/lightning/>`_ as
well as to the `PennyLane documentation <https://docs.pennylane.ai/>`_ for further reference.

**************
Docker support

> **************

Reviewer 2: Is this style compatible with sphinx?

Author: I was following these [guidelines](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#headings).
